cli/src/pcluster/templates/cw_dashboard_builder.py

```diff
@@ -219,6 +228,131 @@ def _add_conditional_storage_widgets(
+                    metric_name=metric_name,
+                    metric_value="1",
+                    default_value=0,
+                )
+            ],
+        )
+        metric_filter.add_depends_on(self.cw_log_group)
+        return metric_filter
+
+    def add_custom_error_metrics(self):
+        """Create custom error metric filter and outputs to cloudwatch graph."""
+        jobs_not_starting_errors = [
+            custom_metric_filter(
+                error_information="Missing IAM Permissions",
+                metric_name="No IAM Group",
+                filter_pattern="UnauthorizedOperation",
+            ),
+            custom_metric_filter(
+                error_information="Missing IAM Permissions",
+                metric_name="Mismatch between IAM Group",
+                filter_pattern="AccessDeniedException",
+            ),
+            custom_metric_filter(
+                error_information="Deep learning AMI invalid",
+                metric_name="AMI larger than root volume",
+                filter_pattern="is smaller than snapshot",
+            ),
+            custom_metric_filter(
+                error_information="Inability to launch due to limits",
+                metric_name="Vcpu limit",
+                filter_pattern="VcpuLimitExceeded",
+            ),
+            custom_metric_filter(
+                error_information="Inability to launch due to limits",
+                metric_name="Volume Limit",
+                filter_pattern="VolumeLimitExceeded",
+            ),
+            custom_metric_filter(
+                error_information="Insufficient Capacity ",
+                metric_name="Node Capacity Insufficient",
+                filter_pattern="InsufficientInstanceCapacity",
+            ),
+        ]
+
+        custom_script_errors = [
+            custom_metric_filter(
+                error_information="Failure in retrieving custom actions",
+                metric_name="Cannot retrieve custom script",
+                filter_pattern="error occurred 403 when calling the HeadObject operation Forbidden",
+            ),
+            custom_metric_filter(
+                error_information="Error in Running Custom Actions",
+                metric_name="Error With Custom Script",
+                filter_pattern="Bootstrap failed with error Failed to run bootstrap recipes",
+            ),
+            custom_metric_filter(
+                error_information="Script Timeout",
+                metric_name="Infinite Custom Script",
+                filter_pattern="systemd udevd worker timeout kill it",
+            ),
+        ]
+
+        compute_node_termination_errors = [
+            custom_metric_filter(
+                error_information="EC2 termination of compute nodes out of PCluster control",
+                metric_name="Terminated EC2 compute node before job submission ",
+                filter_pattern="WARNING Node state check no corresponding instance in EC2 for node",
```

Comment:
Are you sure this can get the error?
I believe you are missing a `-` after `WARNING`

Reply:
i get an error if i include an alphanumeric character

Reply:
Without including the `-` does the metric filter still work?

Reply:
You can write something like:
```
r"WARNING \- Node state check no corresponding instance in EC2 for node"
```

Reply:
You are right. The syntax of filter pattern is different from Python Regex.
According to the doc https://docs.aws.amazon.com/AmazonCloudWatch/latest/logs/FilterAndPatternSyntax.html, your pattern should work